src/charm.py

```diff
@@ -118,6 +132,47 @@ def __init__(self, *args):
             self._certificates.on.certificate_expiring, self._on_certificate_expiring
         )
 
+    def _configure_amf(self, _ : EventBase) -> None:
+        """Handle Juju events.
+
+        Whenever a Juju event is emitted, this method performs a couple of checks to make sure that
+        the workload is ready to be started. Then, it configures the AMF workload,
+        runs the Pebble services and expose the service information through charm's interface.
+
+        Args:
+            _ (EventBase): Juju event
+        """
+        if not self.ready_to_configure():
+            logger.info("The preconditions for the configuration are not met yet.")
+            return
+
+        if not self._private_key_is_stored():
+            self._generate_private_key()
+
+        if not self._csr_is_stored():
+            self._request_new_certificate()
+
+        provider_certificate = self._get_current_provider_certificate()
+        if not provider_certificate:
+            return
+
+        if certificate_update_required := self._is_certificate_update_required(
+            provider_certificate
+        ):
+            self._store_certificate(certificate=provider_certificate)
+
+        desired_config_file = self._generate_amf_config_file()
+        if config_update_required := self._is_config_update_required(desired_config_file):
+            self._push_config_file(content=desired_config_file)
+
+        should_restart = config_update_required or certificate_update_required
+        self._configure_pebble(restart=should_restart)
+
+        try:
+            self._set_n2_information()
+        except ValueError:
+            return
```

Comment:
Why is the method raising a ValueError? Shouldn't this be caught earlier?

Reply:
I realise this was there before, feel free to ignore this comment

Reply:
If the Metallb is not enabled, this method raises ValueErrıor and  on_collect_status charm status  is set to Blocked.